lib/internal/streams/whatwg.js

```diff
@@ -0,0 +1,3336 @@
+        controller[kState].pullAgain = false;
+        readableByteStreamControllerCallPullIfNeeded(controller);
+      }
+    },
+    (error) => readableByteStreamControllerError(controller, error));
+}
+
+function readableByteStreamControllerError(controller, error) {
+  const {
+    stream,
+  } = controller[kState];
+  if (stream[kState].state !== 'readable')
+    return;
+  readableByteStreamControllerClearPendingPullIntos(controller);
+  resetQueue(controller);
+  readableByteStreamControllerClearAlgorithms(controller);
+  readableStreamError(stream, error);
+}
+
+function readableByteStreamControllerCancelSteps(controller, reason) {
+  readableByteStreamControllerClearPendingPullIntos(controller);
+  resetQueue(controller);
+  const result = controller[kState].cancelAlgorithm(reason);
+  readableByteStreamControllerClearAlgorithms(controller);
+  return result;
+}
+
+function readableByteStreamControllerPullSteps(controller, readRequest) {
+  const {
+    pendingPullIntos,
+    queue,
+    queueTotalSize,
+    stream,
+  } = controller[kState];
+  assert(readableStreamHasDefaultReader(stream));
+  if (queueTotalSize) {
+    assert(!readableStreamGetNumReadRequests(stream));
+    const {
+      buffer,
+      byteOffset,
+      byteLength,
+    } = queue.shift();
+    controller[kState].queueTotalSize -= byteLength;
+    readableByteStreamControllerHandleQueueDrain(controller);
+    const view = new Uint8Array(buffer, byteOffset, byteLength);
+    readRequest[kChunk](view);
+    return;
+  }
+  const {
+    autoAllocateChunkSize,
+  } = controller[kState];
+  if (autoAllocateChunkSize !== undefined) {
+    try {
+      const buffer = new ArrayBuffer(autoAllocateChunkSize);
+      pendingPullIntos.push({
+        buffer,
+        byteOffset: 0,
+        byteLength: autoAllocateChunkSize,
+        bytesFilled: 0,
+        elementSize: 1,
+        ctor: Uint8Array,
+        type: 'default',
+      });
+    } catch (error) {
+      readRequest[kError](error);
+      return;
+    }
+  }
+  readableStreamAddReadRequest(stream, readRequest);
+  readableByteStreamControllerCallPullIfNeeded(controller);
+}
+
+function setupReadableByteStreamController(
+  stream,
+  controller,
+  startAlgorithm,
+  pullAlgorithm,
+  cancelAlgorithm,
+  highWaterMark,
+  autoAllocateChunkSize) {
+  assert(stream[kState].controller === undefined);
+  if (autoAllocateChunkSize !== undefined) {
+    assert(Number.isInteger(autoAllocateChunkSize));
+    assert(autoAllocateChunkSize > 0);
+  }
+  controller[kState] = {
+    byobRequest: null,
+    closeRequested: false,
+    pullAgain: false,
+    pulling: false,
+    started: false,
+    stream,
+    queue: [],
+    queueTotalSize: 0,
+    highWaterMark,
+    pullAlgorithm,
+    cancelAlgorithm,
+    autoAllocateChunkSize,
+    pendingPullIntos: [],
+  };
+  stream[kState].controller = controller;
+
+  PromisePrototypeThen(
+    PromiseResolve((async () => startAlgorithm())()),
```

Comment:
```suggestion
    (async () => startAlgorithm())()),
```
there's no need to PromiseResolve the return of an async function